### PR TITLE
Channel: Render age restricted channels

### PR DESCRIPTION
**Pull request by ChunkyProgrammer**
closes https://github.com/iv-org/invidious/issues/3513

This PR:
- gets thumbnail and channel name from the initial request
- gets videos, shorts and streams via autogenerated channel playlists
image

Test Url: `/channel/UCbfnHqxXs_K3kvaH-WlNlig`

### CODE REVIEW

File: src/invidious/channels/about.cr [hunk 1]
```diff
@@ -43,36 +44,97 @@ def get_about_info(ucid, locale) : AboutChannel
...
+    is_family_friendly = false
+    is_age_gated = true
+    tab_names = ["videos", "shorts", "streams"]
+    auto_generated = false
   else
-    author = initdata["metadata"]["channelMetadataRenderer"]["title"].as_s
-    author_url = initdata["metadata"]["channelMetadataRenderer"]["channelUrl"].as_s
-    author_thumbnail = initdata["metadata"]["channelMetadataRenderer"]["avatar"]["thumbnails"][0]["url"].as_s
-    author_verified = has_verified_badge?(initdata.dig?("header", "c4TabbedHeaderRenderer", "badges"))
+    if auto_generated
```
Comment (SamantazFox) [inline src/invidious/channels/about.cr:63]: can you use `elsif` here? That'd greatly reduce the diff's size!

Comment (ChunkyProgrammer) [reply src/invidious/channels/about.cr:63]: It should be possible but lines _97_ to _136_ are outside of the nested `if-else` so it'd require another if statement

Comment (SamantazFox) [reply src/invidious/channels/about.cr:63]: Ohhh, right, forgot about that, thanks!

File: src/invidious/channels/about.cr [hunk 2]
```diff
@@ -45,45 +46,101 @@ def get_about_info(ucid, locale) : AboutChannel
...
-
-    # Raises a KeyError on failure.
-    banners = initdata["header"]["interactiveTabbedHeaderRenderer"]?.try &.["banner"]?.try &.["thumbnails"]?
-    banner = banners.try &.[-1]?.try &.["url"].as_s?
+  if ageGate = initdata.dig?("contents", "twoColumnBrowseResultsRenderer", "tabs", 0, "tabRenderer", "content", "sectionListRenderer", "contents", 0, "channelAgeGateRenderer")
+    description_node = nil
+    author = ageGate["channelTitle"].as_s
+    ucid = initdata["responseContext"]["serviceTrackingParams"][0]["params"][0]["value"].as_s
+    author_url = "https://www.youtube.com/channel/" + ucid
+    author_thumbnail = ageGate["avatar"]["thumbnails"][0]["url"].as_s
```
Comment (SamantazFox) [inline src/invidious/channels/about.cr:58]:
```suggestion
    ucid = initdata.dig("responseContext", "serviceTrackingParams", 0, "params", 0, "value").as_s
    author_url = "https://www.youtube.com/channel/#{ucid}"
    author_thumbnail = ageGate.dig("avatar", "thumbnails", 0, "url").as_s
```

**Review by SamantazFox — CHANGES_REQUESTED** [review]

**Review by SamantazFox — APPROVED** [review]

**Comment by ChunkyProgrammer** [comment]
rebased and fixed merge conflicts, everything seems to be working again

**Review by SamantazFox — APPROVED** [review]

**Comment by SamantazFox** [comment]
Thanks for your contribution ^^

**Pull request merged**